engine/src/main/scala/cromwell/services/MetadataServiceActor.scala

```diff
@@ -0,0 +1,65 @@
+package cromwell.services
+
+import akka.actor.Actor
+import cromwell.engine.workflow.WorkflowManagerActor.WorkflowMetadata
+import cromwell.engine.{ExecutionEventEntry, WorkflowId}
+import cromwell.engine.backend.JobKey
+import cromwell.services.MetadataServiceActor._
+import cromwell.webservice.WorkflowMetadataResponse
+import org.joda.time.DateTime
+
+object MetadataServiceActor {
+
+  sealed trait MetadataServiceActorMessage
+  case class SetStartTime(jobKey: JobKey, startTime: DateTime)
+  case class SetEndTime(jobKey: JobKey, time: DateTime)
+  case class AddExecutionEvent(jobKey: JobKey, executionEventEntry: ExecutionEventEntry)
+  /*
+  We'd have a "set" message for each of these:
```

Comment:
Something that makes clear that a value has changed at a specified point in time, without clobbering whatever current value might be set.  Something like:

```scala
sealed trait MetadataEvent[V] {
  def workflowId: WorkflowId
  def jobKey: JobKey
  def eventName: String
  def eventTime: DateTime
  def value: V
}

/** Every instance of this received by the MetadataServiceActor is simply appended to the DB table; previously existing rows
    for this workflowId + jobKey are not modified.  This retains the full record of what changed and when.  `submission`, `start`, and `end` might
    be derived from the appropriate rows with eventName = "WorkflowStatus".  */
final case class WorkflowStatusEvent(workflowId: WorkflowId, jobKey: JobKey, eventTime: DateTime, value: String) extends MetadataEvent[String] {
  override def eventName = "Workflow Status"
}

```